server/routes/listorder.js

```diff
@@ -6,19 +6,30 @@ const require = createRequire(import.meta.url);
 const router = express.Router();
 
 router.get('/', (req, res) => {
+  const reqCustomerId = req.query.customerId;
   var allData = [];
 
   const getOrderData = async function () {
     try {
       //get db connection
       let pool = await sql.connect(dbConfig);
-
-      let orderResults = await pool
-        .request()
-        .query(
-          '  SELECT orderId,orderDate, ordersummary.customerId, customer.firstName,customer.lastName, totalAmount ' +
-            'from ordersummary left join customer on ordersummary.customerId = customer.customerId order by orderId'
-        );
+      let orderResults;
+      if (reqCustomerId) {
+        orderResults = await pool
+          .request()
+          .input('reqCustomerId', sql.VarChar, reqCustomerId)
+          .query(
+            'SELECT orderId,orderDate, ordersummary.customerId, customer.firstName,customer.lastName, totalAmount ' +
+              'from ordersummary left join customer on ordersummary.customerId = customer.customerId WHERE ordersummary.customerId = @reqCustomerId order by orderId'
+          );
+      } else {
+        orderResults = await pool
+          .request()
+          .query(
+            '  SELECT orderId,orderDate, ordersummary.customerId, customer.firstName,customer.lastName, totalAmount ' +
+              'from ordersummary left join customer on ordersummary.customerId = customer.customerId order by orderId'
+          );
+      }
```

Comment:
At this point I'd move these to separate functions and clean up this whole file by removing the inline async function definition, which is a Ramonism. But, it's not a problem for this milestone.